lib/galaxy/webapps/galaxy/buildapp.py

```diff
@@ -200,7 +207,7 @@ def app_factory(global_conf, load_app_kwds=None, **kwargs):
         if th.is_alive():
             log.debug("Prior to webapp return, Galaxy thread %s is alive.", th)
     # Return
-    return webapp
+    return webapp, app
```

Comment:
Maybe I'm wrong, but can't we access `app` from `webapp`? And if we can, and if the pair as such is only used in `fast_app.py`, maybe give `fast_factory.py` a webapp and let its `factory` function extract the app object - and then pass both of them on to `fast_app.py`? (That way we postpone handling 2 different app objects until they are both needed)
I think the `app_factory` function as it is defined now is more straightforward: *it does one thing*: it builds a thing (a webapp) and returns it - i.e., it's simple. `app_pair`, on the other hand, returns 2 things that are not  meant to be used as a pair - it's just less straightforward, I think. 
I may be wrong though.

Reply:
The WSGI application (`webapp`) has no app, that's associated at each request cycle and app is only passed in for for the the security helper and some config values.

Reply:
> Maybe I'm wrong, but can't we access app from webapp? 

It is all wrapped up in the a bunch of middleware and I couldn't find anyway to extract app systematically from that. I definitely tried.

Reply:
Oh, I see. Then it's fine. Thanks.